pyasic/config/__init__.py

```diff
@@ -149,18 +165,22 @@ def from_am_modern(cls, web_conf: dict) -> "MinerConfig":
 
     @classmethod
     def from_am_old(cls, web_conf: dict) -> "MinerConfig":
+        """Constructs a MinerConfig object from web configuration for old versions of Antminers."""
         return cls.from_am_modern(web_conf)
 
     @classmethod
     def from_goldshell(cls, web_conf: dict) -> "MinerConfig":
+        """Constructs a MinerConfig object from web configuration for Goldshell miners."""
         return cls(pools=PoolConfig.from_am_modern(web_conf))
 
     @classmethod
     def from_inno(cls, web_pools: list) -> "MinerConfig":
+        """Constructs a MinerConfig object from web configuration for Innosilicon miners."""
         return cls(pools=PoolConfig.from_inno(web_pools))
 
     @classmethod
     def from_bosminer(cls, toml_conf: dict) -> "MinerConfig":
+        """Constructs a MinerConfig object from TOML configuration for BOSminer."""
```

Comment:
This constructs from the bosminer.toml file, same as the `to_bosminer` dumps a dict for writing to that file as toml.

Reply:
did you mean `as_bosminer` instead of `to_bosminer`?

Reply:
Yes, as_bosminer.